overlord/stateengine.go

```diff
@@ -117,6 +122,35 @@ func (se *StateEngine) StartUp() error {
 	return nil
 }
 
+func (se *StateEngine) maybeRestartHeldTasks() error {
```

Comment:
this logic again belongs to the overlord/restart package, maybe it should be a real Manager and then this belongs to its StartUp code

Reply:
Done now